routerule/rule_lib.go

```diff
@@ -104,7 +112,8 @@ func RulesMatchSrcFWMark(r, p *Rule) bool {
 		(r.nlRule.Invert == p.nlRule.Invert) &&
 		(r.nlRule.Mark == p.nlRule.Mark) &&
 		(r.nlRule.Mask == p.nlRule.Mask) &&
-		(r.nlRule.Src.String() == p.nlRule.Src.String())
+		((r.nlRule.Src != nil && p.nlRule.Src != nil && r.nlRule.Src.String() == p.nlRule.Src.String()) ||
```

Comment:
If Src is a net.IP then it has an Equal() method; any reason not to use that?  (It handles corner cases like an IPv4 being stored in an IPv6-length slice)

Reply:
These are IPNet which do not have an Equal method.  I added a helper method to ip to compare ipNet pointers.